addons/sourcemod/scripting/shop/item_manager.sp

```diff
@@ -576,6 +578,27 @@ public int ItemManager_EndItem(Handle plugin, int numParams)
 	plugin_item[0] = '\0';
 }
 
+public int ItemManager_UnregisterItem(Handle plugin, int numParams)
+{
+	char item[SHOP_MAX_STRING_LENGTH];
+	int item_id = GetNativeCell(1);
+
+	Format(item, sizeof(item), "%i", item_id);
+
+	h_KvItems.Rewind();
+	if(!h_KvItems.JumpToKey(item))
+		ThrowNativeError(SP_ERROR_NATIVE, "Item id %s is invalid", item);
+
+	DataPack dpCallback = view_as<DataPack>(h_KvItems.GetNum("callbacks", 0));
+	if(dpCallback != null)
+		delete dpCallback;
+	
+	OnItemUnregistered(item_id);
+	
+	h_KvItems.DeleteThis();
+	h_KvItems.Rewind();
```

Comment:
I think there are no reason to rewind after we made our actions there.
We trying to keep Rewind each time we are working with KeyValue items.
@R1KO

Reply:
> I think there are no reason to rewind after we made our actions there.
> We're trying to keep Rewind before each time we are working with KeyValue items.
> @R1KO

during tests, I did a dump (sm_items_dump) of items, without `Rewind()` it was saved somehow crookedly, most likely these are the tricks of `DeleteThis()`...